web/client/src/components/form/TokenForm.jsx

```diff
@@ -30,8 +36,35 @@ export default function TokenForm() {
         tokenAudience: "",
         tokenCredentials: "",
       }}
-      onSubmit={async () => {
-        await sleep(3000);
+      onSubmit={async (values) => {
+        const tokenCred = JSON.parse(values.tokenCredentials);
+        let finalCredentials;
+        if (tokenCred["web"] !== undefined) {
+          finalCredentials = tokenCred["web"];
+        } else {
+          finalCredentials = tokenCred;
+        }
+        let userScopes;
```

Comment:
I think you should treat Audience as a separate parameter field instead of piggybacking on top of "userScopes" and "scopes". In future extension, we will have additional parameters such as "quota project" for the STS token type, which will be used at the same time as "audience". I left a similar comment in PR#5 for Yousef to update the label strategy as well.